### DISCUSSION TITLE
是否可以做到登录用户评论无需审核，匿名用户需要审核

### DISCUSSION THREAD

**Discussion by AnzhiZhang** [walinejs/waline#1563] [Q&A] [answered]
## 背景
计划将评论的用户系统直接使用我的网站的用户系统，如 #1557。我的用户系统用户为可信任用户，是发放的账号。

## 需求
当已登录用户评论时，可以免审核直接发布。未登录的匿名用户评论时，需要审核

**Comment by lizheming** [comment, marked as answer]
用 [preSave](https://waline.js.org/reference/server.html#presave-comment) hooks 可以自行实现：

```
preSave(comment) {
  const { userInfo } = this.ctx.state;
  comment.status = think.isEmpty(userInfo) ? 'waiting' : 'approved';
}
```

**Reply by Maxlife-God** [reply]
您好，请问这段hook在docker部署的情况下如何使用，我在容器的 /app/node_modules/@waline/vercel/config.js 中写了如下代码，但没有效果：
```
module.exports = {
  async preSave(comment) {
  const { userInfo } = this.ctx.state;
  comment.status = think.isEmpty(userInfo) ? 'waiting' : 'approved';
},
};
```

**Reply by lizheming** [reply]
https://github.com/walinejs/waline/blob/main/packages/server/vanilla.js#L17 映射到 `/app/config.js` 即可，不用映射到 node_modules 里面

---
https://github.com/walinejs/waline/blob/main/packages/server/vanilla.js#L17 can be mapped to `/app/config.js` instead of mapping to node_modules
